src/app/shared/wysiwyg/wysiwig.component.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { Selection } from 'prosemirror-state';
+import { cit, create, By } from '../../../testing';
+import { WysiwygComponent } from './wysiwyg.component';
+import { StoryModel } from '../model/story.model';
+import { HalDoc } from '../../core/cms/haldoc';
+
+describe('WysiwygComponent', () => {
+
+  create(WysiwygComponent, false);
+
+  cit('shows markdown formatted as html', (fix, el, comp) => {
+    comp.name = 'description';
+    comp.model = {description: '**bold text**'};
+    fix.detectChanges();
+    expect(el.query(By.css('strong'))).toBeDefined();
+    expect(el).toContainText('bold text');
+  });
+
+  cit('adds image option once images have been loaded', (fix, el, comp) => {
+    comp.name = 'description';
+    comp.model = {description: ''};
+    spyOn(comp, 'insertImageItem');
+    comp.images = [{filename: 'TestImage.jpg'}];
+    fix.detectChanges();
+    expect(comp.insertImageItem).toHaveBeenCalled();
+  });
+
+  cit('doesn\'t add empty links', (fix, el, comp) => {
```

Comment:
I couldn't to get a test to work adding a non empty link, so this isn't exactly a lie. I didn't want to stub out prosemirror because it is a lot, so testing was kind of a disappointment.